x/gamm/pool-models/stableswap/pool.go

```diff
@@ -161,7 +161,11 @@ func (p *Pool) updatePoolLiquidityForExit(tokensOut sdk.Coins) {
 }
 
 func (p *Pool) updatePoolForJoin(tokensIn sdk.Coins, newShares sdk.Int) {
+	numTokens := p.NumAssets()
 	p.PoolLiquidity = p.PoolLiquidity.Add(tokensIn...)
+	if len(p.PoolLiquidity) != numTokens {
+		panic("updatePoolForJoin changed number of tokens in pool")
```

Comment:
I would add `numTokens` and `p.PoolLiquidity` to the panic string via `Sprintf` for better debuggability .